overlord/snapstate/snapmgr.go

```diff
@@ -40,14 +40,38 @@ type SnapManager struct {
 	runner *state.TaskRunner
 }
 
+type SnapSetupFlags Flags
+
+// backward compatibility: upgrade old flags based on snappy.* flags values
+// to Flags if needed
+// XXX: this can be dropped and potentially the type at the earliest
+// in 2.0.9 (after being out for about two prune cycles), at the
+// latest when we need to recover the reserved unusable flag values,
+// or this gets annoying for other reasons
+func (ssfl *SnapSetupFlags) UnmarshalJSON(b []byte) error {
```

Comment:
yes, it's tested by the pair of tests in flagscompat_test.go: adding these comments there for clarity

 
 func (s *flagscompatSuite) TestSnapSetupNewValuesUnchanged(c *C) {
+       // test that new snapstate flags based SnapSetup.Flags is
+       // unmarshalled as is
+

-func (s *flagscompatSuite) TestSnapSetupInterimValsUpgrade(c *C) {
+func (s *flagscompatSuite) TestSnapSetupInterimValuesUpgrade(c *C) {
+       // test that the old snappy.* flags based SnapSetup.Flags
+       // are updated correctly to the new snapstate single shared flag set
        st := state.New(nil)
        st.Lock()
        defer st.Unlock()